### DISCUSSION TITLE
Interfaces on Structs

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7425] [Q&A]
Is there a way to make a single method that essentially "merges" the two definitions below?

```
public static IEnumerable<T> EmptyIfNull<T>(this IEnumerable<T>? This) ...
public static IEnumerable<T> EmptyIfNull<T>(this ImmutableArray<T>? This) ...
```
I'd ideally like it to work with all nullable structs that implement the interface, not just ImmutableArray.

**Comment by CyrusNajmabadi** [comment]
`public static IEnumerable<T> EmptyIfNull<T, TEnumerable>(this TEnumerable enumerable) where TEnumerable : IEnumerable<T>`

That would work for both.  Unfortunately, type inference will not, so you'll have to be extremely explicit.  To the point that i doubt it would be worth it.
